src/Categories/Adjoint/Construction/Adjunctions.agda

```diff
@@ -0,0 +1,163 @@
+{-# OPTIONS --without-K  --allow-unsolved-metas #-}
+
+open import Level
+open import Categories.Category.Core using (Category)
+open import Categories.Category
+open import Categories.Monad
+
+module Categories.Adjoint.Construction.Adjunctions {o ℓ e} {C : Category o ℓ e} (M : Monad C) where
+
+private
+  module C = Category C
+  module M = Monad M
+  open C
+
+open import Categories.Adjoint
+open import Categories.Functor
+open import Categories.Morphism hiding (_≅_)
+open import Categories.Functor.Properties
+open import Categories.NaturalTransformation.Core
+open import Categories.NaturalTransformation.NaturalIsomorphism
+open import Categories.NaturalTransformation.Equivalence renaming (_≃_ to _≅_)
+open import Categories.Morphism.Reasoning as MR
+open import Categories.Tactic.Category
+
+-- the category of adjunctions splitting a given monad
+record SplitObj : Set (suc o ⊔ suc ℓ ⊔ suc e) where
+  constructor splitobj
+  field
+    D : Category o ℓ e
+    F : Functor C D
+    G : Functor D C
+    adj : F ⊣ G
+    GF≃M : G ∘F F ≃ M.F
+
+  module D = Category D
+  module F = Functor F
+  module G = Functor G
+  module adj = Adjoint adj
+  module GF≃M = NaturalIsomorphism GF≃M
+
+  field
+    η-eq : ∀ {A} → GF≃M.⇐.η _ ∘ M.η.η A ≈ adj.unit.η _
+    μ-eq : ∀ {A} → GF≃M.⇐.η _ ∘ M.μ.η A ≈ G.₁ (adj.counit.η (F.₀ A)) ∘ GF≃M.⇐.η _ ∘ M.F.F₁ (GF≃M.⇐.η A)
+
+  {-
+  η-eq:
+                 adj.unit.η A
+            A ------------------+
+    M.η.η A |                   |
+            v                   v
+            MA --------------> GFA
+                  GF≃M.⇐.η A
+
+  μ-eq:
+             GF≃M.⇐.η              G.F₁ (F.F₁ (GF≃M.⇐.η))
+        ---------------------> GFMA -------------------->
+    MMA ---------------------> MGFA --------------------> GFGFA
+     |   M.F.F₁ (GF≃M.⇐.η A)              GF≃M.⇐.η         |
+     |                                                      |
+     | M.μ A                                                | G.₁ (adj.counit.η (F.₀ A))
+     v                                                      v
+     MA -------------------------------------------------> GFA
+                     GF≃M.⇐.η A
+  -}
+
+record Split⇒ (X Y : SplitObj) : Set (suc o ⊔ suc ℓ ⊔ suc e) where
+  constructor split⇒
+  private
+    module X = SplitObj X
+    module Y = SplitObj Y
+
+  field
+    H : Functor X.D Y.D
+    Γ : H ∘F X.F ≃ Y.F
+
+  module Γ = NaturalIsomorphism Γ
+  module H = Functor H
+
+  field
+    μ-comp : ∀ {x : Obj} → Y.GF≃M.⇐.η x ≈ Y.G.F₁ (Γ.⇒.η x) ∘ ((Y.G.F₁ (Functor.F₁ H (X.adj.counit.η (X.F.F₀ x)))) ∘ Y.G.F₁ (Γ.⇐.η (X.G.F₀ (X.F.F₀ x))) ∘ (Y.adj.unit.η (X.G.F₀ (X.F.F₀ x)))) ∘ X.GF≃M.⇐.η x
+
+Split : Monad C → Category _ _ _
```

Comment:
(I know this is still a draft, I'm just curious, so I started looking around.)

I believe that `Split` now has 2 monad instances in scope, but called `M`...

Reply:
Thank you, fixed!